### ISSUE TITLE
Ability to add hashtags to lists

### ISSUE THREAD

**Issue by gelav** [tootsuite/mastodon#7692]
User can subscribe to hashtags adding them to lists.

* * * *

- [x] I searched or browsed the repo’s other issues to ensure this is not a duplicate.

**Comment by Cassolotl** [comment]
Related to *option to follow hashtags #1096*!

**Comment by GidiKroon** [comment]
The current way of following subjects via hashtag is by having a pinned search result column with added hashtags. This is a feature of the multi column layout. If we use the mobile layout, or the simplified layout becomes the default, we lose this ability. I don't think the use case is best served by a UI solution anyway, but is the same kind of thing as defining a list of people. So I'd like to bump the above suggestion, to have the ability to add hashtags as well as people in lists. Hopefully the implementation that currently fills the pinned search result column can be transported to the list system.

(actually following hashtags in your home timeline may be a different thing)

**Comment by projectdp** [comment]
It already is possible to follow hashtags, which will appear in your Home timeline, but I second the suggestion to be able to follow hashtags in specific lists. That way one can follow a collection of related tags by opening a view of their own list, a collection of hashtags and/or people.

**Comment by rbruchal** [comment]
+1

**Comment by MoonfirePDX** [comment]
This would be a significant improvement over just pinning followed hashtags as there are often multiple different tags which may cover the same subject. Being able to add them to lists would allow us to pin a single list with all the tags we feel fit a particular subject or theme without having to switch through all of them.

**Comment by brendanjones** [comment]
@MoonfirePDX  are you aware you can add more than one hashtag to one a pinned column? Up to four iirc. Not saying that solves the issue - I still think hashtags should be addable to list - but maybe it gives you a little more functionality until we have hashtags in lists.

**Comment by ghrasko** [comment]
Still in 2023 this would be a useful feature.

**Comment by brendanjones** [comment]
Dear god this repo needs some management. Duplicates everywhere!

**Comment by loremipsum2022** [comment]
+1

**Comment by atrom** [comment]
+1

**Comment by janEntikan** [comment]
Very much yes please. Hashtags with a strong presence are drowning out fringe (and arguably more interesting) tags. This would solve that.

**Comment by ismailghedamsi** [comment]
How is this not yet implemented. Mastodon is completely unusable to me without this

**Comment by hessert** [comment]
Please add the ability to add hastags to lists. This would be a significant improvement, esp. for mobile users. Thank you!

**Comment by kwfer23** [comment]
Agreed, hashtags in lists would make me happy! It would reduce the noise in my home feed and allow me to better organize content in my lists.

Note that the new feature where list content can be excluded from the home feed will probably make the implementation of "hashtags in a list" more complicated (because you have to scan all hashtags in a post before adding the post to the home feed).

The current management location of "followed hashtags" in the web UI is also a bit hidden. Putting them in lists seems more intuitive to me. FWIW, your home feed is really just a special list of people you're following. That "Following List" should also be able to contain hashtags instead of breaking them out into a special section.

**Comment by SHU-red** [comment]
Yes would just be more than awesome!
Mix all kinds of content every individual user wants to cluster in one "feed"

**Comment by theotherlinh** [comment]
would definitely love to see this. It'd be nice to filter out the hashtags in a separate view from follows as it  can get very noisy.

**Comment by RTbecard** [comment]
Is this feature on the road map?  Or is this being left up to apps to do as a client-side feature?

**Comment by shaedrich** [comment]
@RTbecard I don't think, this is really something that _can_ be implemented by 3rd parties as an extension to the existing bookmarks feature and therefore would have to be implemented as an entirly separate, custom feature for each and every Mastodon client, which, of course, doesn't make sense as this makes nonsense of already having a bookmarks feature

**Comment by brendanjones** [comment]
@shaedrich  did you maybe comment on the wrong issue? This is about hashtags not bookmarks.

**Comment by shaedrich** [comment]
@brendanjones No, it's more like a typo—thanks for the heads-up 👍🏻

**Comment by MikeTaylor** [comment]
I would find this amazingly helpful.

Here is my scenario. I follow Liverpool Football Club, which has both a group (@LiverpoolFC@a.gup.pe) and a hashtag (#LFC). Some people use one, some use the other, so for full coverage I made a list that follows both of these (as well as a few individual I know post who relevant things but don't use the hashtag or the group).

Because I can't add the hashtag to my list, posts that use the hashtag are not for that reason included in the list, and so are not removed from the main timeline. But because most posts to the group _also_ use the hashtag — e.g. `Amazing volley from Núñez! #LFC @LiverpoolFC@a.gup.pe` — the same posts show up in my list (because it includes the group) and in my main timeline (because of the hashes).

If I could add the hashtag to my list, then presumably the hashtagged posts would appear _only_ in the group and not in my main timeline — allowing me the partition I want between my Liverpool-supporting self and my everything-else self.

I hope that's clear!

**Comment by jakkarth** [comment]
I took an initial look at the code for this. There seem to be several different feed types right now. The tag feed is based on PublicFeed and would only consider posts that are public, so adding lists to tag feeds probably isn't the direction to go. Adding tags to lists looks more workable. To pursue this, my approach would be to add a cross-reference table for lists<->tags the same way we have for lists<->accounts, update the lists/ api endpoints to allow for creating and removing those cross-references, update the FeedManager and related worker classes to push statuses with matching tags into the feed for the list, and that should get them displaying in the UI under the list timeline column. I don't have as much experience working on the ui side, so creating an interface that would allow users to add/remove tags from the list in the list edit modal and elsewhere in the web interface would be a challenge for me, but not insurmountable.

That said, I'm more inclined to make a PR for this functionality against glitch than vanilla mastodon. Glitch accepted the PR  made to allow server admins to configure how many hashtags a user could follow in a single timeline column, while vanilla would not. This feels similar to me. But if someone from the vanilla team wants to correct me, I'd be happy to give it a shot and submit the PR here.

**Comment by jakkarth** [comment]
@shaedrich negative. I am talking about a single list that would contain both accounts and tags (or multiple such lists, I suppose). Statuses that are either originating from an account on the list *or* are tagged with a tag on the list would be included in the timeline column. So, you'd end up with a mix of both in a single column. Separate columns is the current state of affairs. Mixed lists is my goal.

**Comment by shaedrich** [comment]
@jakkarth Okay, that's good, then.

Using a separate table for each list item type sure is an option, but I wonder if turning the existing table into a polymorphic one would be the more future-proof solution 🤔

**Comment by MikeTaylor** [comment]
@jakkarth Thank you so much for looking into this!

**Comment by jakkarth** [comment]
Maybe upstream will see my PR to glitch and decide to backport it. Maybe the glitch devs will merge it into glitch. Maybe it will be ignored for several months like my last PR was. Time will tell.

**Comment by loremipsum2022** [comment]
> Maybe upstream will see my PR to glitch and decide to backport it. Maybe the glitch devs will merge it into glitch. Maybe it will be ignored for several months like my last PR was. Time will tell.

Fingers crossed! 🤞🏻

**Comment by shaedrich** [comment]
> Using a separate table for each list item type sure is an option, but I wonder if turning the existing table into a polymorphic one would be the more future-proof solution 🤔

@jakkarth What's your thoughts on that?

**Comment by jakkarth** [comment]
@shaedrich my intention with this PR is to be as minimally invasive as possible, and to follow existing design patterns. Mastodon makes extensive use of the support in rails for defining many-to-many relationships via a cross-reference table, so that's the pattern I elected to follow. If someone else wants to implement it a different way, they're free to make their own PR ;)

**Comment by shaedrich** [comment]
@jakkarth Thanks for the explanation! Yeah, that makes sense. I still don't think, it's the _best_ solution but at least, it's minimally invasive and follows existing design patterns 👍🏻  Thanks for creating the PR, though. It sets the hare running 👍🏻

**Comment by beaniemcgitface** [comment]
> I would find this amazingly helpful.
> 
> Here is my scenario. I follow Liverpool Football Club, which has both a group (@[LiverpoolFC@a.gup.pe](mailto:LiverpoolFC@a.gup.pe)) and a hashtag (#LFC). Some people use one, some use the other, so for full coverage I want a list that follows both of these (as well as a few individuals I know who post relevant things but don't use the hashtag or the group).
> 
> Because I can't add the hashtag to my list, posts that use the hashtag are not for that reason included in my list, and so are not removed from the main timeline. But because most posts to the group _also_ use the hashtag — e.g. `Amazing volley from Núñez! #LFC @LiverpoolFC@a.gup.pe` — the same posts show up in my list (because it includes the group) and in my main timeline (because of the hashtag).
> 
> If I could add the hashtag to my list, then presumably the hashtagged posts would appear _only_ in the group and not in my main timeline — allowing me the partition I want between my Liverpool-supporting self and my everything-else self.
> 
> I hope that's clear!

This is exactly my issue and context - albeit with Chelsea and #CFC. 😬😬 Allegiances aside, this would be a really useful feature.

**Comment by MikeTaylor** [comment]
@beaniemcgitface Can it be true? An #LFC fan and a #CFC fan agree on something! Truly, Mastodon is a catalyst of unprecedented unity! :-)

**Comment by renchap** [comment]
I am currently looking at our list of features for 4.4, and this is a good candidate. I am *not* saying here that we will do it, but I will explore this topic and see if we can do something. Please note that this will probably require more work around lists (the UI is a mess at the moment) so it may be more work than what you would expect.

**Comment by jakkarth** [comment]
@rencha I've got a PR pending at https://github.com/glitch-soc/mastodon/pull/2675. It won't resolve the mess that things are but it does work at least locally. I'd be happy to point it at this repo instead if that would be useful.

**Comment by timtfj** [comment]
For me this is the difference between being able to follow hashtags and not, short of moving everyone I follow into one big list and treating that as my home feed.

**Comment by Qchristensen** [comment]
+1 to this!  Discoverability is so hard on Mastodon that anything which helps is really useful and hashtags are a big part of that.

**Comment by quicoto** [comment]
Wondering if while at this we could increase the number of hashtags per column. Not sure if its a related issue

**Comment by zotheco** [comment]
I would very much welcome this opportunity.

**Comment by smammy** [comment]
Folks, it's generally cool and good to refrain from commenting "+1" or "me too", or complaining about how unfair it is that your pet feature hasn't been implemented yet. Anyone receiving notifications for this issue will thank you. ❤️

**Comment by Strgnoff** [comment]
> I am currently looking at our list of features for 4.4, and this is a good candidate. I am _not_ saying here that we will do it, but I will explore this topic and see if we can do something. Please note that this will probably require more work around lists (the UI is a mess at the moment) so it may be more work than what you would expect.

Thank you for considering this feature for the next release. May I bring to your attention also this related issue and explain why I find this important.

[#22944 Push notification subscriptions should support more granular / arbitrary policies](https://github.com/mastodon/mastodon/issues/22944)

Both of these together are IMHO vital to improve visibility and help people establish more close-knitted networks with each other. Especially accounts that are smaller and post less frequent have difficulties to gain traction, because followers who use Mastodon only occasionally (like, once per day or even just once per week) will probably have almost all other notifications disabled (nobody wants notification spam on their phone or desktop; we're all busy with work), but would really like to receive instant notifications in case of new postings from select accounts (close friends, trusted partners on lists) or particular niche hashtags.

I personally find this kind of granular control over lists and notifications to be a key feature in making Mastodon more user-friendly and helping to make it accessible for users who are not hardcore social media power users, but who want to ditch closed source networks (like Instagram and Facebook) while still being able to get reliable, if occasional updates from their social circle (of those, who migrated to Mastodon as well, as everyone should).

These are core features which belong in the official web interface and official Android/iOS apps, as those are the interfaces that the vast majority of non-power-users are going to use.

I have recently made the same points in [my own article](https://www.tomfichtner.com/article/open-vs-closed-networks/#mastodon)  about open-source networks where I give a brief introduction to Mastodon to a non-technical audience.

**Comment by anniryynanen** [comment]
I'd really like this feature. I started using Mastodon a couple of weeks ago, and at first I was following all the hashtags I found interesting. Soon I had more stuff in my feed than I had time to go through every day.

I still want to see all of the most important things to me, the people I follow, the posts they boost, the small hashtags that only have a couple of posts per week or day. But I had to unfollow some of the bigger hashtags.

Now on the days I have more time, I have to go through the more popular hashtags one at a time. And they often have significant overlap. It's annoying to scroll through the same posts multiple times and try to keep track of whether or not I've reached the spot I left off yesterday.

I know I can have multiple hashtags in one feed in the advanced UI, but that makes the columns tiny! I don't want to read from a 1/4 screen width column. I guess my problem could also be solved by letting me make those columns double wide when I want to.

P.S. I'm a happy Mastodon user. Keep doing what you do.

**Comment by MikeTaylor** [comment]
Indeed: hashtags in lists is one of those powerful ideas that enables a lot of different good things to happen in elegant ways.

**Comment by naknomum** [comment]
this has been kind of touched on (push notifications and hash tags) but i would love to see the little :bell: icon on hash tags, so that i could choose arbitrary ones to show up in my notifications tab.

**Comment by nileshtrivedi** [comment]
FWIW, some of the problems around topic-wise discovery would be solved by #20802 (showing following hashtags in the sidebar similar to Slack/Discord). This is a much simpler improvement as Mastodon already provides [APIs for listing a user's followed hashtags](https://docs.joinmastodon.org/methods/followed_tags/#get) and so this would only require a change in the client-app, making it much easier

I [was able to implement this feature](https://github.com/h3poteto/whalebird-desktop/pull/5028) in [Whalebird](https://github.com/h3poteto/whalebird-desktop) quite quickly. This did not require any changes in the way lists work.

![image](https://github.com/user-attachments/assets/7ccf0da2-6dbd-45c3-8954-e0cdd1cd3de9)

**Comment by k5123** [comment]
Wow, since 2018... and apparently still not possible. :(

**Comment by torbenragn** [comment]
+1

**Comment by jakeyounglol** [comment]
it's absurd that this feature still hasn't been implemented after 7 years.

**Comment by p37307** [comment]
FYI, I found a work-a-round to this problem by searching for account `@tag-<hashtag>@relay.fedi.buzz`, replacing <hashtag> with the tag you want to follow and add to a list, and then following it.  For example, `@tag-mastoadmin@relay.fedi.buzz`. The account's actor: [relay.fedi.buzz/tag/mastoadmin](https://relay.fedi.buzz/tag/mastoadmin) 

I just follow the hashtag account I want and park it on a list, mostly hidden from my home timeline. My home timeline is much cleaner esp for busy hashtags. fyi, the tag account may be empty if no one on your instance already follows it and will populate after new posts are created.

If you aren't aware, since late 2022 [relay.fedi.buzz](https://relay.fedi.buzz) has operated a relay for smaller instances which acts as a firehose. Your instance doesn't have to follow the relay to pull up the tag accounts.

**Comment by brendanjones** [comment]
A note on using the above technique from @p37307 : it seems you do have to search the tag first in order to 'locate' the tag's user.

So for example for the tag 'climate' you can't just drop in the user url ```https://fosstodon.org/@tag-climate@relay.fedi.buzz```, it'll give you a 404 error. You have to seach ```@tag-climate@relay.fedi.buzz``` first, which will show you the user, which you can then follow and forevermore locate at the url.

**Comment by Openhuman2** [comment]
so its already possible but with an helper account? 
You search for the @tag-yourhashtagyouwanttofollow@relay.fedi.buzz 
account and follow it and add the account to your prefered lists.

But its not in the official mastodon search for hashtags right? 
As the new webinterface greatly improved the usabillity of lists this would be a logical next step to get it used by more people.

I'm very happy for this great workaround.

**Comment by titandrive** [comment]
@brendanjones Does that workaround actually work though? I tried it and no posts actually show up under that account 

<img width="1270" height="980" alt="Image" src="https://github.com/user-attachments/assets/1fd23b92-2091-4a6c-b1cc-91972f2b0677" />

**Comment by poyribn** [comment]
@⁠relay.fedi.buzz seems to no longer exist.
I get Error on this Page:
https://mastodon.social/@tag-art@relay.fedi.buzz

> ### 404
>
> The requested page could not be found. Are you sure the URL in the address bar is correct?

**Comment by p37307** [comment]
> [@brendanjones](https://github.com/brendanjones) Does that workaround actually work though? I tried it and no posts actually show up under that account

It did work for me. After I read your reply last night, I searched for it just as the workaround suggested and followed it.  For a few hours it was blank. When I woke up today, the account was populating with **_new_** posts as they are posted, not previous, old posts under the tag. 

<img width="509" height="807" alt="Image" src="https://github.com/user-attachments/assets/afb7e303-7d01-434a-934e-1116f9eff9da" />

**Comment by p37307** [comment]
> @⁠relay.fedi.buzz seems to no longer exist.
> I get Error on this Page:
> https://mastodon.social/@tag-art@relay.fedi.buzz

@sosasees,  did you search for @tag-art@relay.fedi.buzz first? It shows up on my instance. FYI, once you follow it, it may take awhile for posts to start appearing but once they do, it should stay "fresh." Old posts won't appear, only new. 

<img width="804" height="329" alt="Image" src="https://github.com/user-attachments/assets/6e354d65-3894-4599-97a3-4f3c46cd18c6" />

<img width="588" height="378" alt="Image" src="https://github.com/user-attachments/assets/b42a9469-80f0-480b-ab6b-a706c1f4fa4d" />

**Comment by poyribn** [comment]
Thank you, @p37307 ❤️

I get Error when i press “Go to profile @tag-art@⁠relay.fedi.buzz”
but i find the Profile when i press “Profiles matching tag-art@⁠relay.fedi.buzz”.

I don’t know how Search works and Go doesn’t 🫤

**Comment by p37307** [comment]
@sosasees Once you find the “Profiles matching tag-art@⁠relay.fedi.buzz”, can you follow it? If so, do and hopefully, once things starts posting under that tag it will start populating.

**Comment by brendanjones** [comment]
> [@brendanjones](https://github.com/brendanjones) Does that workaround actually work though? I tried it and no posts actually show up under that account

[titandrive](https://github.com/titandrive) Yes, it still works. As mentioned, only new posts show up. Before you follow the profile, it won't have any posts. And following it won't instantly fill your list, because only new posts will populate into the list. 

Both the profile and the list will only have posts from that hashtag *after* you follow the profile and *after* someone has made a new post using that tag.
